pandas/util/_validators.py

```diff
@@ -356,3 +358,48 @@ def validate_fillna_kwargs(value, method, validate_scalar_dict_value=True):
         raise ValueError("Cannot specify both 'value' and 'method'.")
 
     return value, method
+
+
+def validate_setitem_lengths(indexer, value, values):
+    """Validate that value and indexer are the same length.
+
+    An special-case is allowed for when the indexer is a boolean array
+    and the number of true values equals the length of ``value``. In
+    this case, no exception is raised.
+
+    Parameters
+    ----------
+    indexer : sequence
+        The key for the setitem
+    value : array-like
+        The value for the setitem
+    values : array-like
+        The values being set into
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    ValueError
+        When the indexer is an ndarray or list and the lengths don't
+        match.
+    """
+    from pandas.core.indexing import length_of_indexer
+
+    # boolean with truth values == len of the value is ok too
```

Comment:
this is applicable in .where (in internals, but maybe also in frame.py). you *may* want to rename / clean that up.

Reply:
I didn't see anywhere in `Block.where` where similar logic was used.

Reply:
hmm maybe I cleaned this up a while back. IN any event this shouldn't be here at all. _validators is not the correct place. this is purely indexing validation. goes in core/indexing.py

Reply:
I can't really say something about where it belongs (would have to look into more detail), but note that it was copied straight from `internals.py`, and it is still called there. 
So in that sense this PR is just keeping the situation as it was.